test/libsolidity/semanticTests/viaYul/function_entry_checks.sol

```diff
@@ -0,0 +1,29 @@
+// compileViaYul
```

Comment:
Is this the first place where we define the preamble on top of the file? I would be more in favor of introducing a special "block" for it:
```
contract C {
    function f() public returns (uint) {
    }
}
// ----
// compileViaYul: true
// evmVersion: petersburg
// ----
// f() -> 0
```
The block could also contain valid JSON such that it can be parsed utilizing jsoncpp.

Reply:
evmVersion did it like that, too. We have a similar thing for some yul tests, but your idea is probably cleanest - we could also provide a generic interface to it.

Reply:
From an implementation standpoint, it is a bit difficult to implement that, because we would have to peek after having seen `// ---`. Perhaps use a different separator between source code and options?

Reply:
Since I would not expect too many options, what about:
```
contract C {
    function f() public returns (uint) {
    }
}
// ---- compileViaYul; true, evmVersion: petersburg
// f() -> 0
```